src/current/v25.4/troubleshoot-query-plan-regressions.md

```diff
@@ -0,0 +1,141 @@
+---
+title: Troubleshoot Query Plan Regressions
+summary: Troubleshooting guide for when the cost-based optimizer chooses a new query plan that slows performance.
+keywords: query plan, cost-based optimizer, troubleshooting
+toc: true
+docs_area: manage
+---
+
+This page provides guidance on identifying the source of [query plan]({% link {{page.version.version}}/cost-based-optimizer.md %}) regressions.
+
+If the [cost-based optimizer]({% link {{page.version.version}}/cost-based-optimizer.md %}) chooses a new query plan that slows performance, you may observe an unexpected increase in query latency. There are several reasons that the optimizer might choose a plan that increases execution time. This guide will help you understand, identify, and diagnose query plan regressions using built-in CockroachDB tools.
+
+## Before you begin
+
+- [Understand how the cost-based optimizer chooses query plans]({% link {{page.version.version}}/cost-based-optimizer.md %}) based on table statistics, and how those statistics are refreshed.
+
+## Query plan regressions vs. suboptimal plans
+
+The DB Console's [**Insights** page]({% link {{page.version.version}}/ui-insights-page.md %}) keeps track of [suboptimal plans]({% link {{page.version.version}}/ui-insights-page.md %}#suboptimal-plan). A suboptimal plan is a query plan whose execution time exceeds a certain threshold (configurable with the `sql.insights.latency_threshold` cluster setting) and whose slow execution has caused CockroachDB to generate an index recommendation for the table. Table statistics that were once valid, but which are now stale, can lead to a suboptimal plan scenario. A suboptimal plan scenario does not imply that the query plan has changed, and in fact a failure to change the query plan is often the root problem. The **Insights** page identifies these scenarios and provides recommendations on how to fix them.
```

Comment:
I know the Insights page mentions outdated statistics as a possible cause of a suboptimal plan, but I think it's confusing two different concepts. I could be wrong, but the current Insights page seems entirely focused on index recommendations. As far as I can tell, whether or not stats are stale doesn't currently show up as one of the insights. I think it's good for you to mention this, but I'd make it a separate paragraph.

Reply:
Yes, "suboptimal plan" insight only means that we generated an index recommendation which - if applied - will lead to a better plan than the currently chosen, and it's completely independent from the reason for _why_ we don't choose that better plan right now (chances are it's not the stale statistics, but rather a missing index).